widget/tree.go

```diff
@@ -0,0 +1,791 @@
+package widget
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/canvas"
+	"fyne.io/fyne/driver/desktop"
+	"fyne.io/fyne/internal/cache"
+	"fyne.io/fyne/internal/widget"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/storage"
+	"fyne.io/fyne/theme"
+)
+
+const treeDividerHeight = 1
+
+var _ fyne.Widget = (*Tree)(nil)
+
+// Tree widget displays hierarchical data.
+// Each node of the tree must be identified by a Unique ID.
+type Tree struct {
+	BaseWidget
+	Root     string
+	Selected string
+	Offset   fyne.Position
+
+	Children       func(uid string) (c []string)                         // Return a sorted slice of Children Unique IDs for the given Node Unique ID
```

Comment:
Can we use a name here that better indicates that it refers to IDs and not the actual child nodes or their content?
(discussion in slack but we've not reached any agreement)
Perhaps ChildIDs?